### PR TITLE
Added buttonLabel property

### PR DESCRIPTION
**Pull request by chosta**
+ added buttonLabel property to playerInput interface
+ added default values to all classes that implement buttonLabel (AndOptions, OrOptions, etc.) to be equal to "Save" (as is the current state of the app - all buttons are hardcoded to "Save")
+ changed the initial button (when choosing corporation and initial cards) to be "Start" (see Game.ts)

The way we can change any button title is for any input class to add `input.buttonTitle = "Text"`
Please, let me know if this approach works for you guys so I can go on and change more labels
@vincentneko @alrusdi @nwai90

### CODE REVIEW

**Comment by chosta** [comment]
concerning issue #1095

**Comment by nwai90** [comment]
Looks good @chosta. With this approach we can easily customise labels for specific actions 👍

**Comment by vincentneko** [comment]
Yes it looks good.

**Comment by chosta** [comment]
Ok, I'll work on adding some button labels that I think are most needed at the moment and monitor if players accept this change in a positive way. We can merge after I'm done.

**Comment by chosta** [comment]
I've done a handful of changes concerning button labels so we can increase the usability of the app as we agreed to. Since it's a lot of files, I'll double check all the changes one by one so we can be sure nothing is broken. A couple of points on this though:
1. All the label changes are done by me, trying to make sense of every situation. By saying that I mean that they are a bit "opinionated" and I have no problem with changing them at any time.
2. Some of them might seem like repetition (e.g. "Pass for this generation" - yeilds "Pass" button) but the benefit of not having to be super careful whether you picked to play a card or discard one is much bigger.
3. I'm going to play trough a couple of games now and try to get some feedback from my friends, please check out the changes and lets see where this goes.

I think this is a really good step in making the UI work seemlessly and not overburden the player with thinking about whether they clicked the correct "Save" action! Thanks :)

@vincentneko @ssimeonoff @nwai90

File: src/cards/venusNext/Viron.ts [hunk 1]
```diff
@@ -39,7 +39,8 @@ export class Viron implements ICard, CorporationCard {
         }
  
         return new SelectCard(
-            'Perform again an action from a played card',
+            "Perform again an action from a played card",
+            "Play",
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/venusNext/Viron.ts:43]: Yes, I was actually considering Take action here too :)

File: src/cards/venusNext/VenusSoils.ts [hunk 2]
```diff
@@ -44,7 +44,8 @@ export class VenusSoils implements IProjectCard {
         }
 
         return new SelectCard(
-            'Select card to add 2 microbes',
+            "Select card to add 2 microbes",
+            "Add microbe",
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/venusNext/VenusSoils.ts:48]: That is a typo, thanks for noticing

File: src/cards/venusNext/Stratopolis.ts [hunk 3]
```diff
@@ -50,7 +50,8 @@ export class Stratopolis implements IActionCard, IProjectCard, IResourceCard {
         }
 
         return new SelectCard(
-            'Select card to add 2 floaters',
+            "Select card to add 2 floaters",
+            "Add floater",
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/venusNext/Stratopolis.ts:54]: Agreed

File: src/cards/promo/MoholeLake.ts [hunk 4]
```diff
@@ -52,7 +52,7 @@ export class MoholeLake implements IActionCard, IProjectCard {
           return undefined;
         }
         
-        return new SelectCard("Select card to add microbe or animal", availableCards, (foundCards: Array<ICard>) => {
+        return new SelectCard("Select card to add microbe or animal", "Add animal", availableCards, (foundCards: Array<ICard>) => {
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/promo/MoholeLake.ts:55]: True

File: src/cards/promo/CrashSiteCleanup.ts [hunk 5]
```diff
@@ -31,6 +32,7 @@ export class CrashSiteCleanup implements IProjectCard {
 
         const gain2Steel = new SelectOption(
             "Gain 2 steel",
+            "Gains steel",
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/promo/CrashSiteCleanup.ts:35]: Another typo

File: src/cards/prelude/RobinsonIndustries.ts [hunk 6]
```diff
@@ -20,7 +20,7 @@ export class RobinsonIndustries implements IActionCard, CorporationCard {
     }
     public action(player: Player) {
         let minimum = player.getProduction(Resources.MEGACREDITS);
-        let lowest: Array<SelectOption> = [new SelectOption("Increase MC production 1 step", () => {
+        let lowest: Array<SelectOption> = [new SelectOption("Increase MC production 1 step", "Increase +MC", () => {
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/prelude/RobinsonIndustries.ts:23]: The problem I have here is that the label should be "Increase MC production" which is a long one. I tried to experiment with the + sign to annotate "production" but now that I look at it it does not seem right. I'll go with "Increase production" as you suggest and will see if @ssimeonoff can help us with an icon that represents + / - production for the button.

File: src/cards/ViralEnhancers.ts [hunk 7]
```diff
@@ -22,12 +22,12 @@ export class ViralEnhancers implements IProjectCard {
...
             player: player,
             playerInput: new OrOptions(
-                new SelectOption("Add resource to card " + card.name, () => {
+                new SelectOption("Add resource to card " + card.name, "Add resource", () => {
                     player.addResourceTo(card);
                     this.applyCardBonus(player, game, card, resourceCount - 1);
                     return undefined;
                 }),
-                new SelectOption("Gain plant", () => {
+                new SelectOption("Gain plant", "Gain",() => {
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/ViralEnhancers.ts:30]: Agreed

File: src/cards/Insulation.ts [hunk 8]
```diff
@@ -17,7 +17,7 @@ export class Insulation implements IProjectCard {
 
     public play(player: Player, _game: Game) {
         if (player.getProduction(Resources.HEAT) < 1) return undefined;
-        return new SelectAmount("Select amount of heat production to decrease", (amount: number) => {
+        return new SelectAmount("Select amount of heat production to decrease", "Select", (amount: number) => {
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/Insulation.ts:20]: "Decrease" seems ok

File: src/cards/HiredRaiders.ts [hunk 9]
```diff
@@ -20,24 +20,24 @@ export class HiredRaiders implements IProjectCard {
...
+                new SelectOption("Steal 3 mega credit", "Steal MC", () => {
                     player.megaCredits += 3;
                     return undefined;
                 })
             );
         }
 
         return new OrOptions(
-            new SelectPlayer(game.getPlayers(), "Select player to steal up to 2 steel", (selectedPlayer: Player) => {
+            new SelectPlayer(game.getPlayers(), "Select player to steal up to 2 steel", "Steal STEEL", (selectedPlayer: Player) => {
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/cards/HiredRaiders.ts:35]: I was experimenting with this. "Steal steel" looks confusing, so I tried to make it look more distinct. But consistency is more important in this case, so lowercase it is.

File: src/Player.ts [hunk 10]
```diff
@@ -2013,6 +2033,7 @@ export class Player implements ILoadable<SerializedPlayer, Player>{
             playerInput: input
           });
         }
+        console.log
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/Player.ts:2036]: Oops, that is why we need proper linting :/

File: src/Player.ts [hunk 11]
```diff
@@ -1106,6 +1108,7 @@ export class Player implements ILoadable<SerializedPlayer, Player>{
     private playPreludeCard(game: Game): PlayerInput {
       return new SelectCard(
         "Select prelude card to play",
+        "Select and play",
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/Player.ts:1111]: There are a couple of cases where you have to first select a card and then play it. I tried to handle the UX with a single button with both actions "Select and play". This of course is not optimal and now that I think of "Play" should be fine for now. We just need to prompt the user to select the card somehow. Here are some ideas:
+ disable the play button until a card is selected
+ hide the play button as the description already says "Select card" and show it once selected

File: src/Player.ts [hunk 12]
```diff
@@ -1360,6 +1368,7 @@ export class Player implements ILoadable<SerializedPlayer, Player>{
     private airScrapping(game: Game): PlayerInput {
       return new SelectOption(
         "Air scrapping (" + constants.AIR_SCRAPPING_COST + " MC)", 
+        "Use air scrapping",
```
[the remark this answers is not in the document]
Comment (chosta) [reply src/Player.ts:1371]: Yes, actually this one is not even shown, I have to remove it.

File: src/Player.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (nwai90) [reply src/Player.ts:1111]: That's a good point. Disabling the button until a card is selected seems to make sense here, and would also avoid the white popup window error message

**Comment by nwai90** [comment]
> This looks good.
> @nwai90 or @alrusdi could you please also review this one ?

@vincentneko Sure, I can relook at it in a few hours

**Comment by nwai90** [comment]
@chosta LGTM 👍

There are some instances of inconsistent wording (e.g. `"Add microbe(s)", "Add microbe", "Add microbe"`).

We can tidy those up in a separate PR later to use a single term format (e.g. `"Add microbe(s)`) for ease of localisation.

**Pull request merged**